common/desktop-exports

```diff
@@ -62,6 +62,15 @@ mkdir -p $XDG_DATA_HOME
 export XDG_CACHE_HOME=$SNAP_USER_DATA/.cache
 mkdir -p $XDG_CACHE_HOME
 
+# Set config folder to local path
+export XDG_CONFIG_HOME=$SNAP_USER_DATA/.config
+mkdir -p $XDG_CONFIG_HOME
```

Comment:
Note that xdg user dirs are ~/Documents, ~/Videos and the like, and `.config` is an xdg base dir.

For .config, you definitely want to use $SNAP_USER_DATA like you are here, because `/home/<user>/.config` isn't accessible by the snap. Note that $HOME is set to $SNAP_USER_DATA when this code is run.

If, in the future, you wanted the home as configured via /etc/passwd, you can use:
```
getent passwd `id -u` | cut -d ':' -f 6
```

This is conceivably useful for xdg user dirs, like ~/Documents, for example.

Reply:
@jdstrand excellent tip for getting the home dir.  Yeah, we definately want XDG_CONFIG_HOME to be in $SNAP_USER_DATA.

I'm contemplating using xdg-user-dirs-update to generate the user-dirs.dirs with the path to the user's homedir instead of $HOME.  However, longer term that doesn't help us do away with the need for the "home" interface.